bandwidth/bandwidth-impl/src/main/java/com/duckduckgo/bandwidth/impl/BandwidthWorker.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.bandwidth.impl
+
+import android.content.Context
+import androidx.annotation.WorkerThread
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.work.BackoffPolicy
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.ListenableWorker
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import com.duckduckgo.app.global.plugins.worker.WorkerInjectorPlugin
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.mobile.android.vpn.waitlist.store.AtpWaitlistStateRepository
+import com.duckduckgo.mobile.android.vpn.waitlist.store.WaitlistState.InBeta
+import com.squareup.anvil.annotations.ContributesTo
+import dagger.Module
+import dagger.Provides
+import dagger.multibindings.IntoSet
+import kotlinx.coroutines.CoroutineScope
+import timber.log.Timber
+import java.util.concurrent.TimeUnit
+
+@Module
+@ContributesTo(AppScope::class)
+class BandwidthSchedulerModule {
+    @Provides
+    @IntoSet
+    fun provideBandwidthScheduler(workManager: WorkManager, atpWaitlistRepository: AtpWaitlistStateRepository): LifecycleObserver {
+        return BandwidthScheduler(workManager, atpWaitlistRepository)
+    }
+
+    @Provides
+    @IntoSet
+    fun provideBandwidthWorkerInjectorPlugin(
+        bandwidthCollector: BandwidthCollector
+    ): WorkerInjectorPlugin {
+        return BandwidthWorkerInjectorPlugin(bandwidthCollector)
+    }
+}
+
+class BandwidthWorkerInjectorPlugin(
+    private val bandwidthCollector: BandwidthCollector
+) : WorkerInjectorPlugin {
+
+    override fun inject(worker: ListenableWorker): Boolean {
+        if (worker is BandwidthWorker) {
+            worker.bandwidthCollector = bandwidthCollector
+            return true
+        }
+        return false
+    }
+}
+
+class BandwidthScheduler(
+    private val workManager: WorkManager,
+    private val atpWaitlistRepository: AtpWaitlistStateRepository
```

Comment:
I think you should probably use the `AtpCohortManager` that is in the `:vpn-api` module.
If the user has a non-null cohort, then it has enabled AppTP previously.

The `:vpn` module is effectively the `:vpn-impl` module, and `impl` modules should not depend on other `impl` modules :D